qiskit_nature/algorithms/ground_state_solvers/adapt_vqe.py

```diff
@@ -97,16 +139,19 @@ def _compute_gradients(
         for exc in self._excitation_pool:
             # add next excitation to ansatz
             self._ansatz.operators = self._excitation_list + [exc]
-            # set the current ansatz
-            vqe.ansatz = self._ansatz
-            # evaluate energies
-            parameter_sets = theta + [-self._delta] + theta + [self._delta]
-            energy_evaluation = vqe.get_energy_evaluation(self._main_operator)
-            energy_results = energy_evaluation(np.asarray(parameter_sets))
+            # the ansatz needs to be decomposed for the gradient to work
+            vqe.ansatz = self._ansatz.decompose()
```

Comment:
```suggestion
            vqe.ansatz = self._ansatz
```
This has actually since been fixed in Terra so the decomposition is no longer needed.